wagtailio/core/blocks.py

```diff
@@ -179,3 +183,371 @@ class HomePageBlock(blocks.StreamBlock):
+    description = blocks.RichTextBlock(
+        required=False,
+        features=["bold", "italic", "link"],
+    )
+    cta = CTABlock(required=False)
+
+    class Meta:
+        icon = "tick-inverse"
+        label = "Icon bullet"
+        template = "patterns/components/streamfields/icon_bullet_block.html"
+
+
+class IconBulletsBlock(blocks.StructBlock):
+    icon_bullets = blocks.ListBlock(IconBulletBlock())
+
+    class Meta:
+        template = "patterns/components/streamfields/icon_bullets_list_block.html"
+        label = "Icon bullets"
+
+
+class StandaloneQuoteBlock(blocks.StructBlock):
+    quote = blocks.TextBlock(required=True)
+    author = blocks.RichTextBlock(
+        required=True,
+        features=["link"],
+    )
+    author_image = ImageChooserBlock(required=False)
+
+    class Meta:
+        icon = "openquote"
+        label = "Standalone quote"
+        template = "patterns/components/streamfields/standalone_quote_block.html"
+
+
+class MultipleQuoteBlock(blocks.StructBlock):
+    heading = blocks.TextBlock(required=True)
+    quotes = blocks.ListBlock(StandaloneQuoteBlock(), min_num=2)
+
+    class Meta:
+        icon = "openquote"
+        label = "Multiple quotes"
+        template = "patterns/components/streamfields/multiple_quotes_block.html"
+
+
+class RichTextBlock(blocks.StructBlock):
+    rich_text = blocks.RichTextBlock(
+        required=True,
+        features=["bold", "italic", "h2", "h3", "ol", "ul", "link", "document"],
+    )
+
+    class Meta:
+        icon = "bold"
+        label = "Rich text"
+        template = "patterns/components/streamfields/rich_text_block.html"
+
+class StandaloneCTABlock(blocks.StructBlock):
+    short_description = blocks.TextBlock(required=False, max_length=100)
+    cta = CTABlock()
+
+    class Meta:
+        icon = "arrow-right"
+        label = "Standalone CTA"
+        template = "patterns/components/streamfields/standalone_cta_block.html"
+
+
+class TeaserBlock(blocks.StructBlock):
+    page = blocks.PageChooserBlock(required=False, page_type=["blog.BlogPage"])
+    url_chooser = blocks.URLBlock(required=False)
+    image_for_external_link = ImageChooserBlock(required=False)
+    heading_for_external_link = blocks.TextBlock(required=False)
+    subheading_for_ext_link = blocks.TextBlock(
+        label="Subheading for external link", required=False
+    )
+
+    def clean(self, value):
+        errors = {}
+        struct_value = super().clean(value)
+
+        # We want the logical AND of three values
+        sum_external_link_values = (
+            bool(value.get("image_for_external_link"))
+            + bool(value.get("heading_for_external_link"))
+            + bool(value.get("subheading_for_ext_link"))
+        )
+
+        if not value.get("page") and not value.get("url_chooser"):
+            error = ErrorList(
+                [
+                    ValidationError(
+                        "You must specify a blog post or a URL."
```

Comment:
now that we use `page = blocks.PageChooserBlock(required=False, page_type=["blog.BlogPage"])`, perhaps this should be generic.. e.g. `You must specify a page or a URL.`